src/view/site-film-сard-view.js

```diff
@@ -46,5 +46,46 @@ export default class FilmsCardTemplate extends AbstractView {
     this._callback.editClick();
   }
 
+  #findControlButton = (name) => this.element.querySelector(`button[name=${name}]`)
+
+  setControlButtomClickHandler =  (name) => {
+    this.#findControlButton(name).addEventListener('click', () => this.#controlButtonCallback(name));
+  }
+
+
+  #controlButtonCallback = (name) =>{
+
+    this.#setChangeValue(name);
+    this.#etChangeColorControlButton(name);
+
+  }
+
+
+  #setChangeValue = (value) =>{
+    const name = value;
+    const toggler = this.#task[name];
+    this.#task[name] =!toggler;
+  }
+
+#etChangeColorControlButton = (name) =>{
+  const button = this.#findControlButton(name);
+
+  if(this.#task[name]){
+    button.classList.add('film-card__controls-item--active');
+  }
+  else{
+    button.classList.remove('film-card__controls-item--active');
+  }
+
+}
```

Comment:
по идее можно было бы `classList.toggle использовать`.
или, если не известно, что значения будут синхронны, тогда как у тебя - нормально.

**рекомендация** старайся не использовать **else**. В твоем случае можно было бы так поступить. вычислить по `this.#task[name]` какое имя у метода нужно использовать 'add' или 'remove'


```suggestion

#getMethodName = (name)=> (this.#task[name])?('add'):('remove');

#etChangeColorControlButton = (name) =>{
   this.#findControlButton(name)
   .classList[this.#getMethodName(name)]('film-card__controls-item--active');
}
```

в строке `.classList[this.#getMethodName(name)]('film-card__controls-item--active');`
метод add или remove обозначается не через .точку, а через квадратные скобки. 
да так можно.